code/game/objects/items/shields.dm

```diff
@@ -60,6 +109,74 @@
 	materials = list()
 	resistance_flags = FLAMMABLE
 	block_chance = 30
+	transparent = FALSE
+	max_integrity = 65
+
+/obj/item/shield/riot/buckler/shatter(mob/living/carbon/human/owner)
+	playsound(owner, 'sound/effects/bang.ogg', 50)
+	var/turf/T = get_turf(src)
+	new /obj/item/stack/sheet/mineral/wood(T)
+
+/obj/item/shield/riot/flash
+	name = "strobe shield"
+	desc = "A shield with a built in, high intensity light capable of blinding and disorienting suspects. Takes regular handheld flashes as bulbs."
+	icon_state = "flashshield"
+	item_state = "flashshield"
+	var/obj/item/assembly/flash/handheld/embedded_flash
+
+/obj/item/shield/riot/flash/Initialize()
+	. = ..()
+	embedded_flash = new(src)
+
+/obj/item/shield/riot/flash/attack(mob/living/M, mob/user)
+	. =  embedded_flash.attack(M, user)
+	update_icon()
+
+/obj/item/shield/riot/flash/attack_self(mob/living/carbon/user)
+	. = embedded_flash.attack_self(user)
+	update_icon()
+
+/obj/item/shield/riot/flash/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	. = ..()
+	if (. && !embedded_flash.crit_fail)
+		embedded_flash.activate()
+		update_icon()
+
+
+/obj/item/shield/riot/flash/attackby(obj/item/W, mob/user)
+	if(istype(W, /obj/item/assembly/flash/handheld))
+		var/obj/item/assembly/flash/handheld/flash = W
+		if(flash.crit_fail)
+			to_chat(user, "No sense replacing it with a broken bulb.")
+			return
+		else
+			to_chat(user, "You begin to replace the bulb.")
+			if(do_after(user, 20, target = src))

```

Comment:
```suggestion
			if(do_after(user, 20, target = user))
```
in do_after's the `target` arg is only used for what the progress bar is attached to.